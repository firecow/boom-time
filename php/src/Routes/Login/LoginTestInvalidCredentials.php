<?
declare(strict_types=1);

namespace App\Routes\Login;

use App\Exceptions\FileErrorException;
use App\Exceptions\FileNotFoundException;
use App\Mocks\ContextMock;
use App\Mocks\SuperGlobalsMock;
use App\Router\RouteHandler;
use App\Testing\RegressionTest;

class LoginTestInvalidCredentials extends RegressionTest
{

    /**
     * @return string
     * @throws FileErrorException
     * @throws FileNotFoundException
     */
    public function doTest(): string
    {
        $superGlobals = new SuperGlobalsMock();
        $superGlobals->setHTTPPostValue("username", "notacorrectusername");
        $superGlobals->setHTTPPostValue("password", "B9AF6C72E5CF65DB52A44E495BD027CB87A1E90CB1DDD45AEB17B96DE64D613");
        $ctx = new ContextMock($superGlobals);
        $routeHandler = new RouteHandler($ctx);
        $routeResponse = $routeHandler->run("/rest/login/");
        return $routeResponse->prettyPrint();
    }
}


