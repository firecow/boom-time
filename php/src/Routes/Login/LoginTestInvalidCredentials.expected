HTTP/1.1 422 Unprocessable Entity
Content-Length: 34
Content-Type: text/plain

username or password is incorrect.